n3fit/src/evolven3fit/utils.py

```diff
@@ -0,0 +1,105 @@
+from scipy.interpolate import interp1d
+import numpy as np
+import math
+import pathlib
+import yaml
+import shutil
+
+
+class LhapdfLike:
+    pids_dict = {
+        -6: "TBAR",
+        -5: "BBAR",
+        -4: "CBAR",
+        -3: "SBAR",
+        -2: "UBAR",
+        -1: "DBAR",
+        21: "GLUON",
+        1: "D",
+        2: "U",
+        3: "S",
+        4: "C",
+        5: "B",
+        6: "T",
+        22: "PHT",
+    }
+    pids_order = [
+        "TBAR",
+        "BBAR",
+        "CBAR",
+        "SBAR",
+        "UBAR",
+        "DBAR",
+        "GLUON",
+        "D",
+        "U",
+        "S",
+        "C",
+        "B",
+        "T",
+        "PHT",
+    ]
+
+    def __init__(self, pdf_grid, q20, x_grid):
+        self.pdf_grid = pdf_grid
+        self.q20 = q20
+        self.x_grid = x_grid
+        self.funcs = [
+            interp1d(self.x_grid, self.pdf_grid[pid], kind='cubic')
+            for pid in range(len(self.pids_order))
+        ]
+
+    def xfxQ2(self, pid, x, q2):
+        if not math.isclose(q2, self.q20, rel_tol=1e-6):
```

Comment:
I'm worried about this `isclose`  because I can imagine someone writing `2.72` instead of `2.722` so (unless there is a reason for this choice) it should be either a two-digits check or a exact check.
And, in any case, can't this also be check at the beginning of the computation? (not sure when this happens, but I would be very sad if after 30 minutes the computation failed because I mistakenly wrote `2.73` somewhere.

Reply:
Most likely the best solution is a two-digit check. Anyway, I think this check to be harmless because `self.q20` comes from the q20 in the exportgrid while `q2` comes from the q2 in the theory_card, so at the end they are the same thing. The user is not writing the one nor the other. The check is just to be sure that eko is not asking for a wrong q2 to the initial PDF.